src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -765,13 +765,16 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
 
         $container->setParameter('validator.translation_domain', $config['translation_domain']);
 
-        list($xmlMappings, $yamlMappings) = $this->getValidatorMappingFiles($container);
-        if (count($xmlMappings) > 0) {
-            $validatorBuilder->addMethodCall('addXmlMappings', array($xmlMappings));
+        $files = array('xml' => array(), 'yml' => array(), 'yaml' => array());
+        $this->getValidatorMappingFiles($container, $files);
+        $this->getValidatorMappingFilesFromConfig($config, $files);
+
+        if (!empty($files['xml'])) {
+            $validatorBuilder->addMethodCall('addXmlMappings', array($files['xml']));
         }
 
-        if (count($yamlMappings) > 0) {
-            $validatorBuilder->addMethodCall('addYamlMappings', array($yamlMappings));
+        if (!empty($files['yml']) || !empty($files['yaml'])) {
+            $validatorBuilder->addMethodCall('addYamlMappings', array(array_merge($files['yml'], $files['yaml'])));
```

Comment:
What about calling the method twice instead to avoid the call to `array_merge()`?